### PR TITLE
RFC 2 - Functions of Meower Clients and Servers

### PR DESCRIPTION
**Pull request by MikeDev101**

### CODE REVIEW

File: RFC2.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+1. Have long, private, and secure communications
+2. Keep main server load down
+3. Pull content quicker (utilizing a bunch of user servers as a CDN)
+4. Offer alts
+
+Some minor specifications needed to maintain moderation of all systems as follows:
+1. A moderator of the main server will automatically be a moderator of all user servers.
+2. Given special permission, a user server moderator can be promoted to a main server mod.
+3. All content must be filtered and verified by a minimum amount of user servers running scanners and filters. This will hopefully decrease the chances of malicious and/or obscene content from being shared.
+4. Main server and user server moderators have remote access to the server to perform standard antivirus software, ddos protection, and other misc. network administrative measures to keep services running smoothly.
```
Comment (wgyt) [inline RFC2.md:38]: giving moderators SSH access is a fucking terrible idea, limit SSH access to managers on only the main server

**Review by wgyt — CHANGES_REQUESTED** [review]
this PR needs TONS of work and needs to be reviewed by the Moderators team

File: RFC2.md (hunk 1 above)
Comment (MikeDev101) [reply RFC2.md:38]: I suppose I wasn't exactly clear but I was going after more of a remote web app to interact with the server. But I do see your point.

File: RFC2.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+1. Have long, private, and secure communications
+2. Keep main server load down
+3. Pull content quicker (utilizing a bunch of user servers as a CDN)
+4. Offer alts
+
+Some minor specifications needed to maintain moderation of all systems as follows:
+1. A moderator of the main server will automatically be a moderator of all user servers.
+2. Given special permission, a user server moderator can be promoted to a main server mod.
+3. All content must be filtered and verified by a minimum amount of user servers running scanners and filters. This will hopefully decrease the chances of malicious and/or obscene content from being shared.
```
[the remark this answers is not in the document]
Comment (wgyt) [reply RFC2.md:38]: To be clear, I think that filtering on servers other than main server shouldn't be as strict, and that only private servers should have NSFW content

File: RFC2.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+## Client and user server communication
+Akin to discord servers, users can utilize user servers to:
+
+1. Have long, private, and secure communications
+2. Keep main server load down
+3. Pull content quicker (utilizing a bunch of user servers as a CDN)
+4. Offer alts
+
+Some minor specifications needed to maintain moderation of all systems as follows:
+1. A moderator of the main server will automatically be a moderator of all user servers.
```
[the remark this answers is not in the document]
Comment (wgyt) [reply RFC2.md:37]:
```suggestion
1. A moderator of the main server will become a moderator of all servers that request moderation from the Meower Moderation team```

**Review by wgyt — APPROVED** [review]

**Comment by wgyt** [comment]
@tnix100 and others might object to this as it uses hub based federation, what do you think about this implementation

**Review by tnix100 — COMMENTED** [review]
New doc is a lot better, I like the idea of not relying on a main server as much and having most things controlled by a user's "home server". I also didn't like the idea of giving moderators privellage over 3rd party servers.

**Comment by VelocityDesign** [comment]
Wait so meower is federated now?

**Comment by MikeDev101** [comment]
> Wait so meower is federated now?

This RFC sets the foundation for federation within New Meower.

**Pull request merged**